clients/source.go

```diff
@@ -30,6 +31,7 @@ type SourceClient struct {
 	conn           *grpc.ClientConn
 	grpcSocketName string
 	cmdWaitErr     error
+	wg             sync.WaitGroup
```

Comment:
This should always be a pointer (and initialized somewhere) to defend against any future variable copy issues.